src/tree_walker.js

```diff
@@ -62,7 +62,7 @@ TreeWalker.prototype.nextSibling = function() {
  */
 TreeWalker.prototype.parentNode = function() {
   this.currentNode = this.currentParent;
-  this.currentParent = this.currentParent.parentNode;
+  this.currentParent = /** @type {!Node}*/(this.currentParent.parentNode);
```

Comment:
Yer the types here here are all a bit annoying. Its caused because the parentNode could `null` and further up the type expects a `{!Node}`